### PR TITLE
[dataprotection] Support new operations for MUA

### PR DESCRIPTION
**Pull request by zubairabid**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

`az dataprotection`

The following commands and scenarios now have resourceguard-based MUA protection
`az dataprotection backup-vault update` - Modify Soft Delete and Immutability State
`az dataprotection backup-instance stop-protection` - Stop Protection
`az dataprotection backup-instance suspend-backups - Suspend Backups
`az dataprotection backup-instance restore trigger` - Trigger Restore

Also now supporting shorthands for new RecoveryServices critical operations.

Also added: code cleanup to reduce complexity of resource guard mapping/unlock code.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
dataprotection

**Comment by yonzhan** [comment]
Please fix CI issues.

**Comment by zubairabid** [comment]
CI Issues are fixed, please review and merge this PR @yonzhan @evelyn-ys

File: src/dataprotection/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,16 @@
...
 ===============
 
+1.2.0
++++++
+* The following commands and scenarios now have resourceguard-based MUA protection
+* `az dataprotection backup-vault update` - Modify Soft Delete and Immutability State
+* `az dataprotection backup-instance stop-protection` - Stop Protection
+* `az dataprotection backup-instance suspend-backups` - Suspend Backups
+* `az dataprotection backup-instance restore trigger` - Trigger Restore
+* Also now supporting shorthands for new RecoveryServices critical operations.
```
Comment (evelyn-ys) [inline src/dataprotection/HISTORY.rst:13]: Can you provide the related command here?

File: src/dataprotection/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,16 @@
...
+* Also added: code cleanup to reduce complexity of resource guard mapping/unlock code.
```
[the remark this answers is not in the document]
Comment (zubairabid) [reply src/dataprotection/HISTORY.rst]: Resolved.

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**